### PR TITLE
FEAT: Create Organization

### PR DESCRIPTION
**Pull request by chimaihueze**
This pull request introduces the functionality to create a new organization in the system. It includes modifications to the organization entity, organization dto, command handler, mapper, and controller to support creating organizations through an API endpoint.

####     Command Handler:
        CreateOrganizationCommandHandler:
            Implemented the logic to handle CreateOrganizationCommand.
            Added functionality to set CreatedAt and UpdatedAt properties to the current UTC time before saving the organization to the repository.

####    Mapper Profile:
        OrganizationMapperProfile:
            Added mapping configuration for CreateOrganizationDto to Organization.

####    Controller:
        OrganizationController:
            Added the CreateOrganization endpoint to handle POST requests for creating a new organization.
            Configured the endpoint to return a CreatedAtAction result upon successful creation.

####        CreateOrganizationDto:
            Represents the data transfer object for creating a new organization.
            Contains properties such as Name, Description, Email, etc.

####        OrganizationDto:
            Represents the data transfer object for the organization response.
            Includes properties such as Id, Name, CreatedAt, UpdatedAt, etc.

####    CreateOrganizationCommandHandler:
        Maps the CreateOrganizationDto to the Organization entity.
        Sets the CreatedAt and UpdatedAt fields to DateTime.UtcNow.
        Saves the entity to the repository and returns the mapped OrganizationDto.

####    Controller Endpoint:
        Route: POST /api/v1/organizations
        Request Body: CreateOrganizationDto
        Response: 201 Created with OrganizationDto in the response body.

### Authentication:

        The endpoint is protected by the [Authorize] attribute, which means it requires the user to be authenticated before accessing it.
        Only users with valid authentication tokens (e.g., JWT) will be able to create a new organization.

### Testing:

        Verified the correct creation of an organization and mapping of timestamps.
        Ensured that the CreateOrganizationCommandHandler handles the command correctly and returns the expected result.

### CODE REVIEW

File: src/Hng.Application.Test/Features/Organization/CreateOrganizationShould.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+using Hng.Application.Features.Organisations.Handlers;
+using Hng.Application.Features.Organisations.Mappers;
+using Hng.Domain.Entities;
+using Hng.Infrastructure.Repository.Interface;
+using Moq;
+using Xunit;
+
+namespace Hng.Application.Test.Features.Organisations
+{
+    public class CreateOrganizationCommandHandlerTests
```
Comment (NdubuisiJr) [inline src/Hng.Application.Test/Features/Organization/CreateOrganizationShould.cs:13]: Your test file and your test class have different names. Fix this.

**Review by NdubuisiJr — CHANGES_REQUESTED** [review]
This looks pretty. Doesn't it?

Fix the review comment and the Merge conflict. I'll merge this once you are done.

File: src/Hng.Application.Test/Features/Organization/CreateOrganizationShould.cs (hunk 1 above)
Comment (chimaihueze) [reply src/Hng.Application.Test/Features/Organization/CreateOrganizationShould.cs]: DONE

**Review by NdubuisiJr — APPROVED** [review]

**Comment by NdubuisiJr** [comment]
Looks pretty good

**Pull request merged**